src/Jackalope/Node.php

```diff
@@ -302,7 +310,9 @@ private function parseData($rawData, $update, $keepChanges = false)
                 $this->nodes = $nodesInBackend;
                 $this->originalNodesOrder = null;
             }
-            foreach ($oldProperties as $name => $property) {
+            foreach ($oldPropertyValues as $name => $propertyValues) {
+                // TODO needs an update
+                $property = $this->loadProperty($name);
```

Comment:
this is part of the refresh operation. i think it was about removing newly created properties again if we refresh and they where not saved. so you could just have an $oldProperties with the already loaded properties and loop over those here. if they are newly created, they certainly are loaded.